caravel/views.py

```diff
@@ -1843,25 +1846,41 @@ def csv(self, client_id):
     @expose("/fetch_datasource_metadata")
     @log_this
     def fetch_datasource_metadata(self):
-        # TODO: check permissions
-        # TODO: check if datasource exits
         session = db.session
         datasource_class = SourceRegistry.sources[request.args.get('datasource_type')]
         datasource = (
             session.query(datasource_class)
             .filter_by(id=request.args.get('datasource_id'))
             .first()
         )
-        # SUPPORT DRUID
-        # TODO: move this logic to the model (maybe)
-        grains_choices = [str(grain.name) for grain in datasource.database.grains()]
+        # Check if datasource exists
+        if not datasource:
```

Comment:
this function is defined in BaseView and sharable: [https://github.com/airbnb/caravel/pull/1224/files/1cdc6941c6c9639993c3e2c9402ef89748ceae7e#diff-e9b1e371eebef6df66bd8cde507e5931R59](url)